.github/workflows/dev-container.yml

```
@@ -6,33 +6,187 @@
+
+      - name: Upload test results
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results
+          path: test-results/
+          retention-days: 7
+
+      - name: Generate Allure Report
+        if: always()
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC1046:error:5:1: Couldn't find 'fi' for this 'if'

(shellcheck)